api.planx.uk/inviteToPay/sendPaymentEmail.ts

```diff
@@ -71,17 +66,53 @@ const validatePaymentRequest = async (
     `;
     const client = getClientForTemplate(template);
     const headers = { "x-hasura-payment-request-id": paymentRequestId }
-    const { payment_requests_by_pk: paymentRequest } = await client.request(query, { paymentRequestId }, headers);
+    const { query: { session, ...paymentRequest } } = await client.request<ValidatePaymentRequest>(query, { paymentRequestId }, headers);
 
     if (!paymentRequest) throw Error(`Unable to find payment request: ${paymentRequestId}`);
-    if (paymentRequest?.paid_at) throw Error(`Invalid payment request, already paid`);
-    
-    return paymentRequest;
+    if (paymentRequest?.paidAt) throw Error(`Invalid payment request, already paid`);
+
+    return { session, paymentRequest }
   } catch (error) {
     throw Error(`Unable to validate payment request. ${(error as Error).message}`);
   }
 };
 
-export {
-  sendSinglePaymentEmail
+const getInviteToPayNotifyConfig = async (session: SessionDetails, paymentRequest: PaymentRequest): Promise<InviteToPayNotifyConfig> => ({
+  personalisation: {
+    ...session.flow.team.notifyPersonalisation,
+    id: paymentRequest.sessionId,
+    paymentRequestId: paymentRequest.id,
+    payeeEmail: paymentRequest.payeeEmail,
+    payeeName: paymentRequest.payeeName,
+    agentName: paymentRequest.applicantName,
+    address: (paymentRequest.sessionPreviewData?._address as Record<"title", string>).title,
+    fee: getFee(paymentRequest),
+    projectType: await getHumanReadableProjectType(paymentRequest.sessionPreviewData) || "Project type not submitted",
+    serviceName: convertSlugToName(session.flow.slug),
+    expiryDate: calculateExpiryDate(paymentRequest.createdAt),
+    paymentLink: getPaymentLink(session, paymentRequest),
+  }
+});
+
+const getFee = (paymentRequest: PaymentRequest) => {
+  const toPounds = (pence: number) => pence / 100;
+  const fee = toPounds(paymentRequest.paymentAmount).toLocaleString("en-GB", { style: "currency", currency: "GBP" });
+  return fee;
 };
+
+const getPaymentLink = (session: SessionDetails, paymentRequest: PaymentRequest) => {
+  const {
+    flow: {
+      slug: flowSlug,
+      team: {
+        domain,
+        slug: teamSlug,
+      },
+    },
+  } = session
+  // Use custom domain if available or fall back to PlanX URL
+  const serviceURL = domain ? `https://${domain}/${flowSlug}` : `${process.env.EDITOR_URL_EXT}/${teamSlug}/${flowSlug}`;
+  return `${serviceURL}/pay?paymentRequestId=${paymentRequest.id}`
+};
```

Comment:
thanks for adding correct fallback logic for magic links here and fixing the initial assumption I had around the `domain`!